packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/model/FlutterAuthRule.kt

```diff
@@ -16,10 +16,74 @@
 package com.amazonaws.amplify.amplify_datastore.types.model
 
 import com.amplifyframework.core.model.AuthRule
+import com.amplifyframework.core.model.AuthStrategy
+import com.amplifyframework.core.model.ModelField
+import com.amplifyframework.core.model.ModelOperation
 
 data class FlutterAuthRule(val map: Map<String, Any>) {
+    
+    private val authStrategy : AuthStrategy =
+            indexToAuthStrategy(map["authStrategy"] as Int)
+    private val ownerField : String? = map["ownerField"] as String?
+    private val identityClaim : String? = map["identityClaim"] as String?
+    private val groupClaim : String? = map["groupClaim"] as String?
+    private val groups : List<String> = map["groups"] as List<String>
+    private val groupsField : String? = map["groupsField"] as String?
+    private val operations : List<ModelOperation>? =
+            (map["operations"] as List<Int>?)?.map { index ->
+                indexToModelOperation(index)
+            }
+
+    fun indexToAuthStrategy(index: Int) : AuthStrategy {
+        return when(index){
+            0 -> AuthStrategy.OWNER
+            1 -> AuthStrategy.GROUPS
+            2 -> AuthStrategy.PRIVATE
+            3 -> AuthStrategy.PUBLIC
+            else -> throw Exception("indexToAuthStrategy - invalid index supplied: $index")
+        }
+    }
+
+    fun indexToModelOperation(index: Int) : ModelOperation {
+        return when(index){
+            0 -> ModelOperation.CREATE
+            1 -> ModelOperation.UPDATE
+            2 -> ModelOperation.DELETE
+            3 -> ModelOperation.READ
+            else -> throw Exception("indexToModelOperation - invalid index supplied: $index")
+        }
+    }
+
     fun convertToNativeAuthRule(): AuthRule {
-        TODO("Not yet implemented")
+
+        var builder: AuthRule.Builder = AuthRule.builder()
+                .authStrategy( authStrategy )
+
+        if(groups != null && groups.isNotEmpty()){
+            builder.groups(groups)
+        }
+
+        if(!ownerField.isNullOrEmpty()){
+            builder.ownerField(ownerField)
+        }
+
+        if(!identityClaim.isNullOrEmpty()){
+            builder.identityClaim(identityClaim)
+        }
+
+        if(!groupClaim.isNullOrEmpty()){
+            builder.groupClaim(groupClaim)
+        }
+
+        if(!groupsField.isNullOrEmpty()){
```

Comment:
In the various places where you check `isNullOrEmpty()`, I'm not sure you have to. If you pass a null value to the build methods, won't it just store null (which it was going to do, anyway)? 

You might simplify several files by removing all of the `if` checks (assuming that you indeed can do so.)

Reply:
How would you recommend we do that? 

The builder API does not accept nullable variables.

ex: 
        @NonNull
        public AuthRule.Builder groupClaim(@NonNull String groupClaim) {
            this.groupClaim = (String)Objects.requireNonNull(groupClaim);
            return this;
        }

Reply:
Haha, ah right. I remember making a fuss to ensure it was that way, too. Fair enough. :-D